addons/l10n_vn_edi/wizard/account_move_send.py

```diff
@@ -0,0 +1,155 @@
+        for wizard in self:
+            wizard.l10n_vn_edi_enable = any(move.l10n_vn_edi_state == 'ready_to_send' for move in wizard.move_ids)
+
+    @api.depends('l10n_vn_edi_enable')
+    def _compute_l10n_vn_edi_generate_file_checkbox(self):
+        # E-invoicing is a legal requirement in Vietnam, so we can enable by default.
+        for wizard in self:
+            wizard.l10n_vn_edi_generate_file_checkbox = wizard.l10n_vn_edi_enable
+
+    @api.depends('l10n_vn_edi_enable')
+    def _compute_l10n_vn_edi_send_checkbox(self):
+        # E-invoicing is a legal requirement in Vietnam, so we can enable by default.
+        for wizard in self:
+            wizard.l10n_vn_edi_send_checkbox = wizard.l10n_vn_edi_enable
+
+    # -------------------------------------------------------------------------
+    # ATTACHMENTS
+    # -------------------------------------------------------------------------
+
+    @api.model
+    def _get_invoice_extra_attachments(self, move):
+        # EXTENDS 'account'
+        return super()._get_invoice_extra_attachments(move) + move.l10n_vn_edi_sinvoice_file_id
+
+    @api.depends('l10n_vn_edi_send_checkbox')
+    def _compute_mail_attachments_widget(self):
+        # EXTENDS 'account' - add depends
+        super()._compute_mail_attachments_widget()
+
+    def _needs_sinvoice_placeholder(self):
+        return self.l10n_vn_edi_enable and self.l10n_vn_edi_generate_file_checkbox and not self.l10n_vn_edi_send_checkbox
+
+    @api.model
+    def _need_invoice_document(self, invoice):
+        # EXTENDS 'account'
+        return super()._need_invoice_document(invoice) or (not invoice.l10n_vn_edi_sinvoice_file_id and invoice.l10n_vn_edi_state == 'ready_to_send')
+
+    def _get_placeholder_mail_attachments_data(self, move):
+        # EXTENDS 'account'
+        results = super()._get_placeholder_mail_attachments_data(move)
+
+        # Don't add a placeholder if we already generated the file before
+        if self.mode == 'invoice_single' and self._needs_sinvoice_placeholder() and not move.l10n_vn_edi_sinvoice_file_id:
+            results.append({
+                'id': 'placeholder_sinvoice.json',
+                'name': f'{move.name.replace("/", "_")}_sinvoice.json',
+                'mimetype': 'application/json',
+                'placeholder': True,
+            })
+
+        return results
+
+    @api.model
+    def _hook_invoice_document_before_pdf_report_render(self, invoice, invoice_data):
+        # EXTENDS 'account'
+        super()._hook_invoice_document_before_pdf_report_render(invoice, invoice_data)
+        # Ensure that we still generate the file if 'generate' is un-checked but send is.
+        if (invoice_data.get('l10n_vn_edi_generate') or invoice_data.get('l10n_vn_edi_send')) and invoice.l10n_vn_edi_state == 'ready_to_send':
+            errors = invoice._l10n_vn_edi_check_invoice_configuration()
+            if not errors:
+                json_data = invoice._l10n_vn_edi_generate_invoice_json()
```

Comment:
Hmmm so we the json to both sinvoice provider and client ?

Reply:
No, it's not intended :+1: 
It should only be sent to sinvoice and eventually downloaded if asked. It wouldn't be useful to send it t the client as you said.